### ISSUE TITLE
REST API documentation or Open API Swagger specification

### ISSUE THREAD

**Issue by girgink** [4TUResearchData/djehuty#4]
It looks like Djehuty deviates from the previous API, e.g. using uuid instead of id, empty url even if the record is saved. Can you please provide a link to the REST API documentation, Open API specification, or similar?

**Comment by H-Kwon** [comment]
Additionally, I would like to suggest making the link available in /doc directory.

**Comment by gabrielakuhn** [comment]
Dear @girgink 

I am very sorry for such a late reply. Thank you for opening the issue.

I would like to let you know that we recently reviewed our contribution process. Therefore, I am going to close this issue for now. However, if the topic is still relevant, please feel free to open a new issue. You can refer to the [contribution guidelines](https://github.com/4TUResearchData/djehuty?tab=contributing-ov-file#report-bugs-or-request-features) for more details.

We are happy to have your collaboration and appreciate your interest!

**Comment by kairoaraujo** [comment]
Hi @girgink,

The API is now documented, including an OpenAPI specification, at https://djehuty.4tu.nl/api/. It is generated from the HTTP layer rather than maintained by hand, so it tracks the implementation. The rest of the documentation, including the RDF data model behind the API, is at https://djehuty.4tu.nl/.

The new HTTP layer that produces it is not released yet, so the interactive documentation is not served by the running instances for the moment. Once it lands, each deployment will expose it directly, for 4TU.ResearchData that would be https://data.4tu.nl/api/docs, alongside the machine-readable document you can point a client generator at.

If anything in the specification is unclear or does not match what you observe, please do let us know.

**On the deviations you noticed.** djehuty aims to be backward-compatible with the v2 API, but it identifies records by UUID internally, so `uuid` supersedes the old numeric `id`, and `id` is `null` for records that never had one. That is intentional. On the empty `url`: if you can say which endpoint and what state the record was in, that would help, a draft may legitimately have no value there, but on a published record it would be worth a separate issue.

**Where this is heading.** We are working on the API beyond what exists today, with more consistent HTTP response codes and extended functionality on the list. Nothing is settled yet, but if there are rough edges in the current API that get in your way, this is a good moment to tell us, that feedback is what shapes the design.
